### ISSUE TITLE
SDK for content api ?

### ISSUE THREAD

**Issue by spmsupun** [hotelbeds-sdk/hotel-api-sdk-php#90]
Does SDK has any option to access content api ? for example hotel details, hotels list, rate commend etc.

**Comment by vmavromatis** [comment]
ContentAPI is currently not available in the SDK for PHP, but it's planned to be implemented in the future. Leaving this open until it's implemented.

**Comment by vmavromatis** [comment]
Hi @nathanphan, the recommended workflow for using ContentAPI is as such:  

**Step 1** Download the hotel data for all ~170k hotels
GET https://api.test.hotelbeds.com/hotel-content-api/1.0/hotels?fields=all&from=1&to=1000  
GET https://api.test.hotelbeds.com/hotel-content-api/1.0/hotels?fields=all&from=1001&to=2000
.
.
.
GET https://api.test.hotelbeds.com/hotel-content-api/1.0/hotels?fields=all&from=160001&to=170000

**Step 2** Get all types just once

**Step 3** Cross reference the type codes between Step 1 and Step 2

**Step 4** Run the hotels list again about 1 per week to update to latest using lastUpdateTime parameter. In most cases you don't need to update types as they should stay the same, but feel free to update them if you wish.

Running seperate Hotel Detail GET calls for each hotel is not good practice, and it's not the way most clients integrate. There some niche cases where you can use other methods of course.

I hope this answered your question, feel free to ask me anything else you need!

**Comment by nathanphan** [comment]
Hi @vmavromatis, thanks for the tips

I am currently working like your suggestions :) 

thanks to know that others are doing the same :)

Thanks again for a very quick reply

**Comment by vmavromatis** [comment]
> I have a question about step 2, should I store all the types in local database?

Yes, I believe this is the best solution. I would think at least 2 tables, 1 for hotels and 1 for types and use JOIN or UNION or something similar. :)

**Comment by spmsupun** [comment]
@vmavromatis are you guys planning to implement this yet ? If no I'm am gonna implement this, pull requests are accepting ?

**Comment by vmavromatis** [comment]
Hi @spmsupun, there is already a community implementation of ContentAPI in this branch: https://github.com/hadyhallak/hotel-api-sdk-php, feel free to try it out. We are planning to create an official implementation as well, we will make an announcement when we have it ready.

**Comment by ivanpuga** [comment]
@spmsupun Did you manage to save the information in the database?

**Comment by spmsupun** [comment]
@ivanpuga well i have created a table that only store the required (most necessary ) fields, but i don't store the whole data in the database, instead of if a client requested a hotel i check it in my database first and if is not in my database or if it's outdated i insert it to my database

**Comment by isatrio** [comment]
Is there any other way to download the hotel list to start? Like download sql file or something

Seems like I need to create module to fetch the data first before can use it

**Comment by spmsupun** [comment]
@isatrio they have offline data file in csv format

**Comment by isatrio** [comment]
@spmsupun where to download?

**Comment by spmsupun** [comment]
https://developer.hotelbeds.com/docs/read/apitude_cache

**Comment by isatrio** [comment]
Thank you

**Comment by isatrio** [comment]
@spmsupun how to download it actually? I try to download but no luck. I have tried to do it using http request. Can you please help me explain how to download it?

**Comment by ahmedsaber111** [comment]
Hello,

I know it's irrelevant, but I have difficulty to pull static data via ContentAPI using the API KEY of Hotel Sandbox with maximum daily rate of 50 request.

Do you have any alternative ways to pull the static data, currently I wrote the scripts to do the task but the rate is very limited.

Is there are any alternatives to download the static data as CSV for instance?

**Comment by BrodriguezHbeds** [comment]
Hi @ahmedsaber111,

The plan for the apiKey you obtained when registering in the developer portal is limited, as is intended to serve as an evaluation plan. Our commercial department (apitude@hotelbeds.com) could provide an apiKey on a plan with a more flexible quota.

We do not support hotel static data file downloading in any capacity. 

Regards,
Bruno
